### PR TITLE
Web feature test

### PR DESCRIPTION
**Pull request by RYin-28**
Created a basic header. 
Did not use sample headers from Mantine because their examples integrated Mantine stylings which made it hard to follow.
image

### CODE REVIEW

File: apps/web/src/App.tsx [hunk 1]
```diff
@@ -1,42 +1,20 @@
-import { useEffect, useState } from "react";
-import reactLogo from "./assets/react.svg";
-import viteLogo from "/vite.svg";
-import "./App.css";
+//import { useEffect, useState } from "react";
+//import reactLogo from "./assets/react.svg";
+//import viteLogo from "/vite.svg";
+//import "./App.css";
```
Comment (drmicrochip97) [inline apps/web/src/App.tsx:4]: Remove comments

File: apps/web/src/common/Top_Header.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Main Header and elements
+import { Button, Container, Grid, Group, Title } from '@mantine/core';
+
+
+function Top_Header() {
```
Comment (drmicrochip97) [inline apps/web/src/common/Top_Header.tsx:5]: Change case style to TitleCase
for example
`Top_Header` -> `TopHeader`

File: apps/web/src/main.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import ReactDOM from 'react-dom/client'
 import App from './App.tsx'
-import './index.css'
+//import './index.css'
```
Comment (drmicrochip97) [inline apps/web/src/main.tsx:4]: remove comment

**Review by drmicrochip97 — COMMENTED** [review]
Few more changes

File: apps/web/src/App.tsx [hunk 4]
```diff
@@ -1,42 +1,16 @@
...
+    <MantineProvider theme = {{ colorScheme: 'dark' }} withGlobalStyles withNormalizeCSS>
+      <div className="App">
+        <TopHeader />
       </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-      <p>{message}</p>
+    </MantineProvider>
     </>
```
Comment (drmicrochip97) [inline apps/web/src/App.tsx:14]: You can remove these react fragments `</>`. They're basically divs and aren't needed here.

File: apps/web/src/App.tsx [hunk 5]
```diff
@@ -1,42 +1,16 @@
...
-  const [count, setCount] = useState(0);
-
-  const [message, setMessage] = useState("");
 
-  useEffect(() => {
-    void fetch("/api")
-      .then((res) => res.text())
-      .then(setMessage);
-  }, []);
+function App() {
```
Comment (drmicrochip97) [inline apps/web/src/App.tsx:5]: I prefer having the export attached to the function name, so for example

```ts
export function App() {
```

You have to modify the import in "main.tsx" like so

```ts
import { App } from "./App.tsx";
```

File: apps/web/src/App.tsx [hunk 6]
```diff
@@ -1,42 +1,16 @@
-import { useEffect, useState } from "react";
-import reactLogo from "./assets/react.svg";
-import viteLogo from "/vite.svg";
-import "./App.css";
```
Comment (drmicrochip97) [inline apps/web/src/App.tsx:4]: Make sure to remove any unneeded files. For example, we are not using App.css so we should be able to just remove it. Same goes for any other files.

File: apps/web/src/App.tsx [hunk 7]
```diff
@@ -1,42 +1,16 @@
...
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
+    <MantineProvider theme = {{ colorScheme: 'dark' }} withGlobalStyles withNormalizeCSS>
+      <div className="App">
+        <TopHeader />
```
Comment (drmicrochip97) [inline apps/web/src/App.tsx:11]: Can we remove this className? It doesn't look like it's actually used. It also looks like this div isn't used.

File: apps/web/src/common/TopHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
+// Main Header and elements
+import { Button, Container, Grid, Group, Title } from '@mantine/core';
+
+
+function TopHeader() {
```
Comment (drmicrochip97) [inline apps/web/src/common/TopHeader.tsx:5]: I would also in this instance 
```ts
export function TopHeader() {
```

and like before modify "App.tsx" to import like so
```ts
import { TopHeader } from "./common/TopHeader";
```

File: apps/web/src/common/TopHeader.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  
+  return (
+    <Container size = "fluid">
+      {/* blah */}
```
Comment (drmicrochip97) [inline apps/web/src/common/TopHeader.tsx:9]: nit: Make sure this comment gets removed

File: apps/web/src/common/TopHeader.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+            <Button variant = "subtle" size = "md"> About </Button>
+          </Group>
+        </Grid.Col>
+      </Grid>
+
+    </Container>
+  );
+}
+
+export default TopHeader;
```
Comment (drmicrochip97) [inline apps/web/src/common/TopHeader.tsx:32]: remove

File: apps/web/src/common/TopHeader.tsx [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+          </Title>
+        </Grid.Col>
+
+        <Grid.Col span = "auto" >
+          <Group position = "right" spacing = "xl">
+            <Button variant = "subtle" size = "md"> Home </Button>
+            <Button variant = "subtle" size = "md"> Menu </Button>
+            <Button variant = "subtle" size = "md"> Contact Us </Button>
+            <Button variant = "subtle" size = "md"> About </Button>
+          </Group>
```
Comment (drmicrochip97) [inline apps/web/src/common/TopHeader.tsx:24]: My automatic tool will do this but for right now just remove the white space for " Home ", " Menu ", " Contact Us ", " About ".

Comment (drmicrochip97) [reply apps/web/src/common/TopHeader.tsx:24]: ## Before
```ts
<Button variant = "subtle" size = "md"> Home </Button>
```
## After
```ts
<Button variant = "subtle" size = "md">Home</Button>
```

**Review by drmicrochip97 — CHANGES_REQUESTED** [review]

File: package.json [hunk 12]
```diff
@@ -6,14 +6,20 @@
     "apps/*"
   ],
   "scripts": {
+
```
Comment (drmicrochip97) [inline package.json:9]: remove blank line

File: package.json [hunk 13]
```diff
@@ -6,14 +6,20 @@
...
+    "localforage": "^1.10.0",
+    "match-sorter": "^6.3.1",
+    "react-router-dom": "^6.14.2",
+    "sort-by": "^1.2.0",
+    "tabler-icons-react": "^1.56.0"
   }
-}
+  
+  }
+
```
Comment (drmicrochip97) [inline package.json:25]: fix formatting for this file

File: apps/web/src/libs/menu/menu.tsx [hunk 14]
```diff
@@ -0,0 +1,14 @@
+// Menu page
```
Comment (drmicrochip97) [inline apps/web/src/libs/menu/menu.tsx:1]: Hmm, redundant comment on top you can remove. Same for the other pages

File: apps/web/src/libs/common/top-header/topheader.tsx [hunk 15]
```diff
@@ -0,0 +1,54 @@
+// Main Header and elements
```
Comment (drmicrochip97) [inline apps/web/src/libs/common/top-header/topheader.tsx:1]: remove redundant comments.

https://www.youtube.com/watch?v=Bf7vDBBOBUA

File: apps/web/src/libs/common/top-header/topheader.tsx [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+import { useNavigate } from 'react-router-dom'
+import { Button, Container, Grid, Group, Title } from '@mantine/core'
+
+export function TopHeader() {
```
Comment (drmicrochip97) [inline apps/web/src/libs/common/top-header/topheader.tsx:5]: skewer-case the filename to match the folder `top-header`

File: apps/web/src/libs/common/top-header/topheader.tsx [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+  }
+  const navigateToLocation = () => {
+    navigate('/location')
+  }
+  const navigateToContactUs = () => {
+    navigate('/contact-us')
+  }
+  const navigateToAbout = () => {
+    navigate('/about')
+  }
```
Comment (drmicrochip97) [inline apps/web/src/libs/common/top-header/topheader.tsx:21]: All these functions should be removed in favor of anonymous functions directly in the component since they are really small and simple. For example,

```tsx
<Button variant="subtle" size="md" onClick={() => navigate('/about') }>
   About
</Button>
```

**Review by drmicrochip97 — CHANGES_REQUESTED** [review]

File: apps/web/src/libs/common/top-header/topheader.tsx (hunk 16 above)
Comment (RYin-28) [reply apps/web/src/libs/common/top-header/topheader.tsx:5]: is it fine to keep the function name as `TopHeader`?

Comment (drmicrochip97) [reply apps/web/src/libs/common/top-header/topheader.tsx:5]: Yah, components should be TitleCase

**Review by drmicrochip97 — APPROVED** [review]
